examples/pytorch/pyg_interface.py

```diff
@@ -0,0 +1,203 @@
+    """
+
+    def __init__(self, ge):
+        """Initialize DeepGNN feature store."""
+        super().__init__()
+        self.ge = ge
+
+    def _put_tensor(self, tensor, attr) -> bool:
+        """Put tensor."""
+        return False
+
+    def _remove_tensor(self, attr) -> bool:
+        """Remove tensor."""
+        return False
+
+    def _get_tensor(self, attr):
+        """Get tensor."""
+        feature = np.array([[0, 121]]) if attr.attr_name == "x" else np.array([[1, 1]])
+        return torch.Tensor(
+            self.ge.node_features(attr.index.detach().numpy(), feature, np.float32)
+        ).squeeze()
+
+    def _get_tensor_size(self, attr):
+        return attr.size()
+
+    def get_all_tensor_attrs(self):
+        """Obtain all tensor attributes stored in this :class:`FeatureStore`."""
+        output = []
+        for i in ["x", "y"]:
+            ta = TensorAttr()
+            ta.group_name = "0"
+            ta.attr_name = f"{i}"
+            output.append(ta)
+        return output
+
+
+class DeepGNNGraphStore(GraphStore):
+    """A class to access edges from a DeepGNN graph engine.
+
+    Args:
+        ge: MemoryGraph The graph engine to sample from.
+        node_path: str Path of nodes file to sample initial nodes from.
+    """
+
+    def __init__(self, ge, node_path):
+        """Initialize DeepGNN graph store."""
+        super().__init__()
+        self.ge = ge
+        self.node_ids = np.loadtxt(node_path, dtype=np.int64)
+
+    def _put_edge_index(self, edge_index, edge_attr) -> bool:
+        """Put edge index."""
+        return False
+
+    def _remove_edge_index(self, edge_attr) -> bool:
+        """Remove edge index."""
+        return False
+
+    def _get_edge_index(self, edge_attr):
+        """Get edge index."""
+        edge_type = int(edge_attr.edge_type[1])
+        edge = self.ge.neighbors(self.node_ids, edge_type)
+        srcs = []
+        for i, e in enumerate(edge[-1]):
+            srcs.extend([self.node_ids[i]] * e)
+        edge_index = (
+            torch.Tensor(srcs).long(),
+            torch.Tensor(edge[0]).long(),
+        )
+        return edge_index
+
+    def get_all_edge_attrs(self):
+        """Obtain all edge attributes stored in the :class:`GraphStore`."""
+        output = []
+        ta = EdgeAttr(
+            ("0", "0", "0"),
+            "coo",
+            size=[self.ge.node_count(0), self.ge.node_count(np.array([0, 1, 2, 3]))],
+        )
+        output.append(ta)
+
+        return output
+
+
+def train():
+    """Train the model."""
+    model.train()
+
+    total_loss = total_examples = 0
+    for data in tqdm.tqdm(loader):
+        data = data.to(device)
+        data = data[("0", "0", "0")]
+        optimizer.zero_grad()
+
+        h = model(data.x, data.edge_index)
+
+        h_src = h[data.edge_label_index[0]]
+        h_dst = h[data.edge_label_index[1]]
+        link_pred = (h_src * h_dst).sum(dim=-1)  # Inner product.
```

Comment:
Please, avoid [copy pasting code](https://github.com/pyg-team/pytorch_geometric/blob/02cc18d9d6841ecda4eb61eebb48b86f1aaae477/examples/graph_sage_unsup_ppi.py#L52) from other repos even with source comments

Reply:
The problem is this interface is meant to be a remote backend for PyG so I need to use their model and dataloader and just copy the glue code from an example since it is not importable. What should I do instead?

Reply:
Write train loop on your own that is similar to theirs.